scout/build/case.py

```diff
@@ -175,55 +216,75 @@ def build_case(case_data, adapter):
             )
             continue
         panel = {
+            "display_name": panel_obj["display_name"],
+            "is_default": False,
+            "nr_genes": len(panel_obj["genes"]),
             "panel_id": panel_obj["_id"],
             "panel_name": panel_obj["panel_name"],
-            "display_name": panel_obj["display_name"],
-            "version": panel_obj["version"],
             "updated_at": panel_obj["date"],
-            "nr_genes": len(panel_obj["genes"]),
+            "version": panel_obj["version"],
         }
         if panel_name in default_panels:
             panel["is_default"] = True
-        else:
-            panel["is_default"] = False
         panels.append(panel)
 
-    case_obj["panels"] = panels
+    return panels
 
-    case_obj["dynamic_gene_list"] = []
 
-    # Meta data
-    genome_build = case_data.get("genome_build", "37")
-    if not genome_build in ["37", "38"]:
-        pass
-        ##TODO raise exception if invalid genome build was used
+def set_timestamps(case_obj):
+    """Set case_obj's internal creation and update timestamps to now'"""
+    now = datetime.now()
+    case_obj["created_at"] = now
+    case_obj["updated_at"] = now
 
-    case_obj["genome_build"] = genome_build
 
-    if case_data.get("rank_model_version"):
-        case_obj["rank_model_version"] = str(case_data["rank_model_version"])
+def set_suspects(case_obj, case_data):
+    if case_data.get("suspects"):
+        case_obj["suspects"] = case_data["suspects"]
 
-    if case_data.get("sv_rank_model_version"):
-        case_obj["sv_rank_model_version"] = str(case_data["sv_rank_model_version"])
 
-    if case_data.get("rank_score_threshold"):
-        case_obj["rank_score_threshold"] = float(case_data["rank_score_threshold"])
+def set_causatives(case_obj, case_data):
+    if case_data.get("causatives"):
+        case_obj["causatives"] = case_data["causatives"]
+
+
+def set_assignee(case_obj, case_data):
+    if case_data.get("assignee"):
+        case_obj["assignees"] = [case_data["assignee"]]
 
-    # Cohort information
+
+def set_analysis_date(case_obj, case_data):
+    if case_data.get("analysis_date"):
+        case_obj["analysis_date"] = case_data["analysis_date"]
+
+
+def set_genome_build(case_obj, case_data):
+    genome_build = case_data.get("genome_build", 37)
+    if not genome_build in [37, 38]:
```

Comment:
The of lines are:
```
    genome_build = case_data.get("genome_build", "37")
    if not genome_build in ["37", "38"]:
        pass
        ##TODO raise exception if invalid genome build was used
```

`genome_build` is set to 37 (int) in our main branch. It will enter the if-clause and exit without anything being done because of `pass`. When set to raise an exception the test this will fail, becuase the variable is an int and not a string.

You can try to patch below code and try it out:

```
--- a/scout/build/case.py
+++ b/scout/build/case.py
@@ -195,8 +195,7 @@ def build_case(case_data, adapter):
     # Meta data
     genome_build = case_data.get("genome_build", "37")
     if not genome_build in ["37", "38"]:
-        pass
-        ##TODO raise exception if invalid genome build was used
+        raise ValueError(f"Genom build {genome_build} not supported")

     case_obj["genome_build"] = genome_build
```

Reply:
Since build HAS to be a string (either "37" or "38") we could perhaps also fix the pydantic parsing of the case config file and validate it with something like this:

if "37" in genome_build:
     case["build"]="37"
elif "38" in genome_build:
     case["build"]="38"
else:
     crash